notebooks/MAST/HSC/HSC_TAP/HSC_TAP.ipynb

```diff
@@ -0,0 +1,472 @@
+    "SELECT SourceID, ImageID, SourceRA, SourceDec, D, Filter, Detector, MagAper2, StartMJD\n",
+    "FROM dbo.DetailedCatalog\n",
+    "WHERE \n",
+    "   MatchID={}\n",
+    "   AND Detector='ACS/WFC' AND Filter='F475W' AND Det='Y'\n",
+    "ORDER BY StartMJD\n",
+    "\"\"\".format(matchid))\n",
+    "HSC_details = job.get_results()\n",
+    "# HSC_details.sort('StartMJD')\n",
+    "HSC_details.pprint()\n",
+    "\n",
+    "pylab.rcParams.update({'font.size': 16})\n",
+    "pylab.figure(1,(10,6))\n",
+    "pylab.scatter(HSC_details['StartMJD'], HSC_details['MagAper2'])\n",
+    "pylab.xlabel('MJD')\n",
+    "pylab.ylabel('A_F475W')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "***\n",
+    "### Use Case: Create a color magnitude diagram for the Small Magellanic Cloud\n",
+    "<div style=\"text-align: right\">[(top)](#title_ID)</div>\n",
+    "\n",
+    "For another example of using data from a TAP service, we start by doing a search around the SMC with a .25 degree radius for objects with ACS F555W and F814W measurements. HSC TAP will limit us to 100k responses by default. Note this is a large query that can take over a minute to run. See <a href=\"http://archive.stsci.edu/hst/hsc/help/use_case_2_v3.html\" target=\"_blank\">HSC Use Case 2</a> for more details."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "t0 = time.time()\n",
+    "\n",
+    "job = HSC_service.launch_job_async(\"\"\"\n",
+    "SELECT MatchID, MatchRA, MatchDec, CI, A_F555W, A_F814W\n",
+    "FROM dbo.SumMagAper2CatView\n",
+    "WHERE A_F555W_N > 0 and A_F814W_N > 0\n",
+    "    AND CONTAINS(POINT('ICRS', MatchRA, MatchDec),CIRCLE('ICRS',13.1866,-72.8286,0.25))=1\n",
+    "   \"\"\")\n",
+    "HSC_results = job.get_results()\n",
+    "print(\"Query completed in {:.1f} sec\".format(time.time()-t0))\n",
+    "HSC_results.pprint()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, plot the color-magnitude diagram for the ~100k points retrieved from the database. This uses kernel density estimate for the crowded plot, which can take a few minutes to complete."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "f555w = HSC_results['A_F555W']\n",
+    "f814w = HSC_results['A_F814W']\n",
+    "VminusI = f555w-f814w\n",
+    "CI = HSC_results['CI']\n",
+    "w = numpy.where((CI>0.9) & (CI<1.6) & (VminusI > -1.5) & (VminusI < 1.5))\n",
+    "print(len(w[0]),\"points remaining after CI and V-I filtering\")\n",
+    "\n",
+    "# Calculate the point density\n",
+    "x = numpy.array(VminusI[w])\n",
+    "y = numpy.array(f555w[w])\n",
+    "xy = numpy.vstack([x,y])\n",
+    "\n",
+    "t0 = time.time()\n",
+    "z = gaussian_kde(xy)(xy)\n",
```

Comment:
As we discussed in person @theresadower, this is taking a while to run, and slows down the builds quite a bit.  And the longest delay is indeed this one.

However, I think I have an easy fix: the plot *looks* basically the same even if we thin this data set a lot.  So you could do something like:

```suggestion
    "z = gaussian_kde(xy[:, ::25])(xy)\n",
    "#z = gaussian_kde(xy)(xy)\n", #uncomment this line to do the KDE on the full dataset
```

Then add some words in the text above that say something like "we work only with 1/25th of the data so that the notebook executes quickly, but you can uncomment the commented-out line for the full plot".

Reply:
Forwarding this to Rick White, the author of the plot and use case, will tag here once we have his GitHub information, or make changes based on his recommendation.

Reply:
Changes made for using the thinned data and commenting/uncommenting lines for full plot generation.

@eteq: thoughts on fast_kde? The thinned version runs quite fast without another install so I'm inclined to use it for now, but we could add text about it as an option.

Reply:
Ah, re: `fast_kde`: I'm not opposed to that as long as it's not a long/complicated/error-prone install.  The way to try it is probably to go into the `.travis.yml` file and underneath the `CONDA_DEPENDENCIES` line, add:
```
- PIP_DEPENDENCIES="fast_kde"
```
and then we can see how the build goes.  Also add it to the `requirements.txt` (eventually we'll use that to determine dependencies, but until then, the `PIP_DEPENDENCIES` trick is the way to go.

Reply:
@rlwastro : in my conda environment, my pip install of fast_kde is failing ("could not find a version that satisfies the requirement fast_kde (from versions: ) No matching distribution found for fast_kde). There is also a fastKDE which pip can find but is also failing installation on my system. Is there something else I should be doing for this install?  I'm uncomfortable checking in the code requiring this when I can't get it to work locally.

Reply:
@theresadower Try fastkde instead of fast_kde?
Here is a link to the [PyPi](https://pypi.org/project/fastkde/) repository.